File: .github/actions/label-cleanup/action.rb
require 'octokit'

repo = ENV["GITHUB_REPOSITORY"]
label = ENV["LABEL"]
comment = ENV["COMMENT"]
token = ENV["TOKEN"]

client = Octokit::Client.new(:access_token => token)
client.auto_paginate = true

open_issues = client.list_issues(repo, { :labels => label, :state => 'open'})

open_issues.each do |issue|
    client.add_comment(repo, issue.number, comment)
    client.remove_label(repo, issue.number, label)
    client.close_issue(repo, issue.number)
end

File: .github/actions/label-cleanup/action.js
const core = require("@actions/core");
const github = require("@actions/github");

async function run() {
    try {
        const label = core.getInput("label");
        const comment = core.getInput("comment");
        const token = core.getInput("token");

        const octokit = github.getOctokit(token);

        const issues = await octokit.rest.issues.listForRepo({
            labels: label,
            state: "open",
            owner: github.context.repo.owner,
            repo: github.context.repo.repo,
        });

        issues.data.forEach(async (issue) => {
            await octokit.rest.issues.createComment({
                body: comment,
                issue_number: issue.number,
                owner: github.context.repo.owner,
                repo: github.context.repo.repo,
            });

            await octokit.rest.issues.removeLabel({
                name: label,
                issue_number: issue.number,
                owner: github.context.repo.owner,
                repo: github.context.repo.repo,
            });

            await octokit.rest.issues.update({
                issue_number: issue.number,
                owner: github.context.repo.owner,
                repo: github.context.repo.repo,
                state: "closed",
            });
        });
    } catch (error) {
        core.setFailed(error.message);
    }
}

run();
